config.go

```diff
@@ -409,12 +409,26 @@ func (cfg *Config) unshadow() {
 }
 
 func generateSwarmSecret() (string, error) {
-	key := make([]byte, 32)
-	_, err := crand.Read(key)
+	const secretLength = 64
+
+	// get random data
+	rand := make([]byte, secretLength)
+	_, err := crand.Read(rand)
 	if err != nil {
 		return "", fmt.Errorf("Error reading from rand: %v", err)
 	}
-	return hex.EncodeToString(key), nil
+
+	const charStart = 33
+	const charEnd = 126
+	const numChars = charEnd - charStart + 1
+
+	// modify random chars to be in desired range
+	secret := make([]byte, secretLength)
+	for j := range rand {
+		secret[j] = byte(int(rand[j])%numChars + charStart)
+	}
+
+	return string(secret), nil
```

Comment:
Ah, yes, very very good point. Thanks for catching that! Was going to have @hsanjuan check this out, but you got there first ;)